File: mail/common/src/main/java/com/fsck/k9/mail/AuthenticationFailedException.java

package com.fsck.k9.mail;

public class AuthenticationFailedException extends MessagingException {
    public static final long serialVersionUID = -1;

    public AuthenticationFailedException(String message) {
        super(message);
    }

    public AuthenticationFailedException(String message, Throwable throwable) {
        super(message, throwable);
    }
}

File: mail/common/src/main/java/com/fsck/k9/mail/AuthenticationFailedException.kt
package com.fsck.k9.mail

import net.thunderbird.core.common.exception.MessagingException

class AuthenticationFailedException @JvmOverloads constructor(
    message: String,
    throwable: Throwable? = null,
    val messageFromServer: String? = null,
) : MessagingException(message, throwable) {
    val isMessageFromServerAvailable = messageFromServer != null
}
